### DISCUSSION TITLE
Grouping rows together

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#1145] [General]
I'm going to display my shallow SQL knowledge here. Maybe this is simply an SQL question...

I have a table that contains property values and tax bills for this year and previous years. I would like to build a query that displays some statistics (average change in price between last year and this, average change in tax bills, etc) and outputs a table with rows for separate ranges of property values:
   0-$100K 
   $100K - $250K
   $250K - $500K
   $500K - $1M
   ... etc.

I don't really know how to make such a single query in SQL, but I wondered if PRQL offers any facility for doing this.

Or is this simply magical thinking? :-)

**Comment by max-sixty** [comment]
I don't think there's anything in PRQL that would make this especially terse. 

Indeed, until we have https://github.com/prql/prql/issues/504, forming the groups will likely require an s-string with a `CASE`.

So, yes, I think this is probably more of a SQL question — create a column with the bins, then do the aggregations grouping by the bins.

Sorry not to be more helpful!

**Reply by richb-hanover** [reply]
Yeah... Given that there are only six or so "bands of prices", it's probably the most efficient use of my time to run it as several queries, each with the desired range, e.g. ~~`filter [ price > 100000, price <= 249999]`~~ `filter ( price > 100000) and (price <= 249999)`

Thanks.

**Reply by richb-hanover** [reply]
It looks as if the new-ish switch/case statement https://github.com/prql/prql/issues/1286 might do what I want... I'll horse around.

**Reply by max-sixty** [reply]
Yes good point! That will! (It's still evolving though, so expect changes...)
